src/commandHelpers/send-helper.ts

```diff
@@ -0,0 +1,190 @@
+    return sendTransactionStruct;
+}
+
+export async function sendAndWaitTransactionConfirmations(transaction: ITransactionData) {
+    this.actionStart("Sending transaction");
+    const sendResponse = await this.api.sendTransaction(transaction);
+    this.actionStop();
+    if (sendResponse.errors) {
+        throw new Error(sendResponse.errors);
+    }
+
+    this.actionStart("Waiting for transaction confirmation");
+    const transactionFromNetwork = await this.waitTransactionConfirmations(
+        this.api.getBlockTime(),
+        transaction.id,
+        1,
+        1,
+    );
+    this.actionStop();
+    return transactionFromNetwork;
+}
+
+export function checkAndGetAmount(amountArg: string, isSatoAmount: boolean) {
+    // Check amount
+    const amountParts = amountArg.replace(" ", "").split(".");
+
+    if (amountArg.includes(",") || amountParts.length > 2) {
+        throw new Error("Please, only use a decimal point for the amount");
+    }
+
+    if (amountParts.length === 2 && amountParts[1].length > 8) {
+        throw new Error(
+            "The number of significant digits displayed after floating point numbers has to be lower or equal to 8",
+        );
+    }
+
+    const amount = Number.parseFloat(amountArg);
+    if (isNaN(amount) || amount <= 0) {
+        throw new Error("The amount should be a positive number");
+    }
+
+    if (isSatoAmount && !Number.isInteger(amount)) {
+        throw new Error("sato UNS amount has to be an integer");
+    }
+
+    return amount;
+}
+
+export async function resolveWalletAddress(id: string, isRecipient: boolean = true): Promise<string> {
+    let resolvedAddress: string;
+
+    if (id && id.startsWith("@")) {
+        try {
+            const resolveResult = await didResolve(`${id}?*`, this.api.network.name);
+            if (resolveResult.error) {
+                throw resolveResult.error;
+            }
+            resolvedAddress = resolveResult.data as string;
+        } catch (e) {
+            if (e && e.response && e.response.status === 404) {
+                throw new Error(`${isRecipient ? "Recipient" : "Sender"} @unik-name does not exist`);
+            }
+            throw new Error(`${isRecipient ? "Recipient" : "Sender"} @unik-name does not match expected format`);
+        }
+    } else {
+        if (!crypto.validateAddress(id, this.api.getVersion())) {
+            try {
+                resolvedAddress = Address.fromPublicKey(id);
+            } catch (_) {
+                throw new Error(`${isRecipient ? "Recipient" : "Sender"} address does not match expected format`);
+            }
+        } else {
+            resolvedAddress = id;
+        }
+    }
+    return resolvedAddress;
+}
+
+export async function checkAndConfirmWallet(checkWalletExistence: boolean, address: string, passphrase?: string) {
+    if (checkWalletExistence) {
+        this.actionStart("Check recipient existence");
+        const exists = await this.checkWalletExistence(address);
```

Comment:
Where is `checkWalletExistence` function ? 
Who is `this`?

Reply:
Right, fail on refactoring, I'll create an helper with command instance